crates/wasmtime/src/runtime/vm/instance/allocator/pooling.rs

```diff
@@ -652,23 +664,30 @@ unsafe impl InstanceAllocator for PoolingInstanceAllocator {
         ty: &wasmtime_environ::Memory,
         memory_index: Option<DefinedMemoryIndex>,
     ) -> Result<(MemoryAllocationIndex, Memory)> {
-        // FIXME(rust-lang/rust#145127) this should ideally use a version of
-        // `with_flush_and_retry` but adapted for async closures instead of only
-        // sync closures. Right now that won't compile though so this is the
-        // manually expanded version of the method.
-        let e = match self.memories.allocate(request, ty, memory_index).await {
-            Ok(result) => return Ok(result),
-            Err(e) => e,
-        };
+        self.live_memories.fetch_add(1, Ordering::AcqRel);
```

Comment:
I was on the fence about whether to just do `Ordering::Relaxed` here, but if someone ends up using metrics in actual business logic (like server backpressure) it would be a pretty nasty surprise for these to underflow...

Reply:
Oh it's not, the main difference is I saw this here and no one saw it there most likely. There's no need for that to use `AcqRel` either and IMO it should also be using `Relaxed`